### DISCUSSION TITLE
New Panel Type (TCX)

### DISCUSSION THREAD

**Discussion by liptonj** [aqualinkd/AqualinkD#183] [General]
Hello,  I have the new Jandy TCX panel.  I started going through the code and sniffing the panel to try and figure out how this works.   

I am seeing the saltwater % and PPM in the serial log but I am not getting any values on the dashboard.   
Below is a sample from the debug log.    it seems that 0x71 looks like a request status message and 0x79 are the ACK messages 
so I added that to the definition.  If I am not correct please let me know

[aqualinkd.log](https://github.com/sfeakes/AqualinkD/files/8849780/aqualinkd.log)
[RS485.log](https://github.com/sfeakes/AqualinkD/files/8849781/RS485.log)
[RS485raw.log](https://github.com/sfeakes/AqualinkD/files/8849782/RS485raw.log)

Jun-07-22 03:55:04 AM Debug:   RS Serial: To 0xd8 of type       TCX Status | HEX: 0x10|0x02|0xd8|0x71|0x00|0x5b|0x10|0x03|
Jun-07-22 03:55:04 AM Debug:   RS Serial: Serial read 11 bytes
Jun-07-22 03:55:04 AM Debug:   RS Serial: To 0x00 of type          TCX Ack | HEX: 0x10|0x02|0x00|0x79|0x81|0x00|0x00|0x00|0x0c|0x10|0x03|
Jun-07-22 03:55:04 AM Debug:   RS Serial: Serial read 8 bytes
Jun-07-22 03:55:04 AM Debug:   RS Serial: To 0xd8 of type       TCX Status | HEX: 0x10|0x02|0xd8|0x71|0x00|0x5b|0x10|0x03|
Jun-07-22 03:55:04 AM Debug:   RS Serial: Serial read 11 bytes
Jun-07-22 03:55:04 AM Debug:   RS Serial: To 0x00 of type          TCX Ack | HEX: 0x10|0x02|0x00|0x79|0x81|0x00|0x00|0x00|0x0c|0x10|0x03|
Jun-07-22 03:55:05 AM Debug:   RS Serial: Serial read 8 bytes
Jun-07-22 03:55:05 AM Debug:   RS Serial: To 0xd8 of type       TCX Status | HEX: 0x10|0x02|0xd8|0x71|0x00|0x5b|0x10|0x03|
Jun-07-22 03:55:05 AM Debug:   RS Serial: Serial read 11 bytes
Jun-07-22 03:55:05 AM Debug:   RS Serial: To 0x00 of type          TCX Ack | HEX: 0x10|0x02|0x00|0x79|0x81|0x00|0x00|0x00|0x0c|0x10|0x03|
Jun-07-22 03:55:05 AM Debug:   RS Serial: Serial read 8 bytes
Jun-07-22 03:55:05 AM Debug:   RS Serial: To 0xd8 of type       TCX Status | HEX: 0x10|0x02|0xd8|0x71|0x00|0x5b|0x10|0x03|
Jun-07-22 03:55:05 AM Debug:   RS Serial: Serial read 11 bytes
Jun-07-22 03:55:05 AM Debug:   RS Serial: To 0x00 of type          TCX Ack | HEX: 0x10|0x02|0x00|0x79|0x81|0x00|0x00|0x00|0x0c|0x10|0x03|
Jun-07-22 03:55:06 AM Debug:   RS Serial: Serial read 8 bytes
Jun-07-22 03:55:06 AM Debug:   RS Serial: To 0x50 of type            AR %% | HEX: 0x10|0x02|0x50|0x11|0x55|0xc8|0x10|0x03|
Jun-07-22 03:55:06 AM Debug:   RS Serial: Serial read 11 bytes
Jun-07-22 03:55:06 AM Debug:   RS Serial: To 0x00 of type           AR PPM | HEX: 0x10|0x02|0x00|0x16|0x1e|0x00|0x00|0x00|0x46|0x10|0x03|
Jun-07-22 03:55:06 AM Debug:   RS Serial: Serial read 8 bytes
Jun-07-22 03:55:06 AM Debug:   RS Serial: To 0xd8 of type       TCX Status | HEX: 0x10|0x02|0xd8|0x71|0x00|0x5b|0x10|0x03|
Jun-07-22 03:55:06 AM Debug:   RS Serial: Serial read 11 bytes
Jun-07-22 03:55:06 AM Debug:   RS Serial: To 0x00 of type          TCX Ack | HEX: 0x10|0x02|0x00|0x79|0x81|0x00|0x00|0x00|0x0c|0x10|0x03|
Jun-07-22 03:55:06 AM Debug:   RS Serial: Serial read 8 bytes
Jun-07-22 03:55:06 AM Debug:   RS Serial: To 0xd8 of type       TCX Status | HEX: 0x10|0x02|0xd8|0x71|0x00|0x5b|0x10|0x03|
Jun-07-22 03:55:06 AM Debug:   RS Serial: Serial read 11 bytes
Jun-07-22 03:55:06 AM Debug:   RS Serial: To 0x00 of type          TCX Ack | HEX: 0x10|0x02|0x00|0x79|0x81|0x00|0x00|0x00|0x0c|0x10|0x03|
Jun-07-22 03:55:07 AM Debug:   RS Serial: Serial read 8 bytes
Jun-07-22 03:55:07 AM Debug:   RS Serial: To 0xd8 of type       TCX Status | HEX: 0x10|0x02|0xd8|0x71|0x00|0x5b|0x10|0x03|
Jun-07-22 03:55:07 AM Debug:   RS Serial: Serial read 11 bytes
Jun-07-22 03:55:07 AM Debug:   RS Serial: To 0x00 of type          TCX Ack | HEX: 0x10|0x02|0x00|0x79|0x81|0x00|0x00|0x00|0x0c|0x10|0x03|
Jun-07-22 03:55:07 AM Debug:   RS Serial: Serial read 8 bytes
Jun-07-22 03:55:07 AM Debug:   RS Serial: To 0xd8 of type       TCX Status | HEX: 0x10|0x02|0xd8|0x71|0x00|0x5b|0x10|0x03|
Jun-07-22 03:55:07 AM Debug:   RS Serial: Serial read 11 bytes
Jun-07-22 03:55:07 AM Debug:   RS Serial: To 0x00 of type          TCX Ack | HEX: 0x10|0x02|0x00|0x79|0x81|0x00|0x00|0x00|0x0c|0x10|0x03|
Jun-07-22 03:55:08 AM Debug:   RS Serial: Serial read 8 bytes
Jun-07-22 03:55:08 AM Debug:   RS Serial: To 0x50 of type            AR %% | HEX: 0x10|0x02|0x50|0x11|0x55|0xc8|0x10|0x03|
Jun-07-22 03:55:08 AM Debug:   RS Serial: Serial read 11 bytes
Jun-07-22 03:55:08 AM Debug:   RS Serial: To 0x00 of type           AR PPM | HEX: 0x10|0x02|0x00|0x16|0x1e|0x00|0x00|0x00|0x46|0x10|0x03|
Jun-07-22 03:55:08 AM Debug:   RS Serial: Serial read 8 bytes
Jun-07-22 03:55:08 AM Debug:   RS Serial: To 0xd8 of type       TCX Status | HEX: 0x10|0x02|0xd8|0x71|0x00|0x5b|0x10|0x03|
Jun-07-22 03:55:08 AM Debug:   RS Serial: Serial read 11 bytes
Jun-07-22 03:55:08 AM Debug:   RS Serial: To 0x00 of type          TCX Ack | HEX: 0x10|0x02|0x00|0x79|0x81|0x00|0x00|0x00|0x0c|0x10|0x03|
Jun-07-22 03:55:08 AM Debug:   RS Serial: Serial read 7 bytes
Jun-07-22 03:55:08 AM Debug:   RS Serial: To 0x78 of type            Probe | HEX: 0x10|0x02|0x78|0x00|0x8a|0x10|0x03|
Jun-07-22 03:55:08 AM Debug:   RS Serial: Serial read 8 bytes
Jun-07-22 03:55:08 AM Debug:   RS Serial: To 0xd8 of type       TCX Status | HEX: 0x10|0x02|0xd8|0x71|0x00|0x5b|0x10|0x03|
Jun-07-22 03:55:08 AM Error:   AqualinkD: No probe on '0x0a', giving up! (please check config)
Jun-07-22 03:55:08 AM Debug:   RS Serial: Closed serial port

**Comment by liptonj** [comment]
AqualinkD serial_logger V1.4
Notice:  RS Serial: Logging serial information!
Please wait.|

Notice:  RS Serial: RS485 interface received 600 packets in 123 seconds (~4.88 Msg/Sec)
Notice:  RS Serial: Jandy Control Panel Model   : 
Notice:  RS Serial: Jandy Control Panel Version : 
Notice:  RS Serial: Jandy ID's found
Notice:  RS Serial: ID 0xd8 is in use  <-- Master control panel
Notice:  RS Serial: ID 0x50 is in use  <-- Salt Water Generator (Aquarite mode)
Notice:  RS Serial: 

Notice:  RS Serial: Pentair ID's found
Notice:  RS Serial: ID 0x10 is in use  <-- Unknown Device
Notice:  RS Serial:

**Comment by sfeakes** [comment]
I just looked at some of the TCX information and the logs you posted.  This is not a complete control panel, and looks like it’s locked to use a single iAqualink protocol.  So kind of like a modern PDA only control panel.  Since Aqualinkd relies on using keypad protocol (except with PDA only panel), I don’t think you will get very far.  It looks like it would take significant code changes for this to work. 

The reason aqualinkd sees the SWG information is because it reads that directly from the SWG and not the control panel.

**Reply by liptonj** [reply]
Ok, thank you my goal is not as much control but systems status, for example, trying to get the pool temp and salt ppm etc out into my home automation

It seems 0xda is the master control panel where the pump and all the sensors hook up into.  But I am not understanding the status message data as it does not mean anything.   I was able to replicate turning on and off my  JVA with command 0x72.   

I see what you mean by a modern PDA as the interface device connects into 

I am enjoying going through the data and trying to understand this.   So any help in pointing me in the right direction would be great.  Not really looking for you to make the code changes
